src/store/modules/nft.js

```diff
@@ -1006,42 +1028,56 @@ const actions = {
       }))
     );
   },
-  async fetchBookstoreCMSProductsByTagId({ commit }, { tagId, t }) {
+  async fetchBookstoreCMSProductsByTagId(
+    { commit, getters },
+    { tagId, t, offset }
+  ) {
+    if (
+      !tagId ||
+      getters.nftGetBookstoreCMSProductsByTagIdIsFetching(tagId) ||
+      getters.nftGetBookstoreCMSProductsByTagIdIsFetchingMore(tagId)
+    ) {
+      return;
+    }
+
+    const isFetchingMore = !!offset;
+    const fetchingMutation = isFetchingMore
+      ? TYPES.NFT_SET_BOOKSTORE_CMS_PRODUCTS_BY_TAG_ID_IS_FETCHING_MORE
+      : TYPES.NFT_SET_BOOKSTORE_CMS_PRODUCTS_BY_TAG_ID_IS_FETCHING;
     try {
-      commit(TYPES.NFT_SET_BOOKSTORE_CMS_PRODUCTS_BY_TAG_ID_IS_FETCHING, {
-        id: tagId,
-        value: true,
-      });
+      commit(fetchingMutation, { id: tagId, value: true });
 
       const { data } = await this.$api.get(
-        api.fetchBookstoreCMSProductsByTagId(tagId, { t })
+        api.fetchBookstoreCMSProductsByTagId(tagId, { t, offset })
       );
 
       commit(TYPES.NFT_SET_BOOKSTORE_CMS_PRODUCTS_BY_TAG_ID, {
         id: tagId,
         value: data.records,
+        isAppend: isFetchingMore,
+        offset: data.offset,
       });
 
       if (!data.records?.length) {
         // eslint-disable-next-line no-console
         console.warn(`CMS did not return any products for tag: ${tagId}`);
       }
     } finally {
-      commit(TYPES.NFT_SET_BOOKSTORE_CMS_PRODUCTS_BY_TAG_ID_IS_FETCHING, {
-        id: tagId,
-        value: false,
-      });
+      commit(fetchingMutation, { id: tagId, value: false });
     }
   },
+  async fetchMoreBookstoreCMSProductsByTagId({ dispatch, getters }, { tagId }) {
+    const offset = getters.nftGetBookstoreCMSProductsPaginationOffsetByTagId(
+      tagId
+    );
+    if (!offset) return;
```

Comment:
Not sure if we should just let it fetch with `offset = 0` if `nftGetBookstoreCMSProductsByTagIdHasFetched` is not set in this case

Reply:
but `offset` is not a number
<img width="451" alt="image" src="https://github.com/user-attachments/assets/528d43b7-8452-4c48-ad89-3bf287ca1632" />

Reply:
Is the guard trying to guard the case where we have not set offset yet, or the list has ended so there is no offset from API? If latter then no problem